cypress/S3Monitor.py

```diff
@@ -0,0 +1,130 @@
+        RoleArn=role_arn,
+        RoleSessionName='session1',
+    )
+
+    s3Client = boto3.client(
+        's3',
+        aws_access_key_id=result['Credentials']['AccessKeyId'],
+        aws_secret_access_key=result['Credentials']['SecretAccessKey'],
+        aws_session_token=result['Credentials']['SessionToken']
+    )
+
+    return s3Client
+
+# Extract the plus part from emails of the form:
+# basename+pluspart@example.com
+def getPlusPartFromEmailAddress(email):
+    match = re.search("[^\+]\+(.+)@", email)
+    if match is None:
+        return ""
+    return match.group(1)
+
+def parseBody(bodyContent, subject, thetype, linkRegex):
+    regex = 'https:\/\/\S+' + linkRegex + '\/[a-zA-Z0-9]+'
+
+    match = re.search(regex, bodyContent)
+
+    if match is not None: 
+        s = match.start()
+        e = match.end()
+        activationLink = bodyContent[s:e]
+        print(f'{ thetype } link { activationLink }')
+
+        emailRegex = 'To: (.+)'
+
+        emailMatch = re.search(emailRegex, bodyContent)
+        if emailMatch is not None:
+            es = emailMatch.start()
+            ee = emailMatch.end()
+            toEmail = bodyContent[es:ee]
+
+            userId = getPlusPartFromEmailAddress(toEmail)
+            print(f'userId {userId}')
+            contents = f'{toEmail[:-1]},{activationLink}'
+            filePath = f'{activation_emails_path}/{userId}.{thetype}'
+            emailFile = open(filePath,'w')
+            emailFile.write(contents)
+            emailFile.close()
+    else:
+        print(f'Message: {subject} does not match regex {regex}') 
+        print('----------------------------------------------------------------------------------')
+
+def parse_email(bodyContent, s3Key):
+    activate_subject = 'Activate your lasting power of attorney account'
+    reset_password_subject = 'Password reset request'
+    reset_password_no_account_subject = 'Request to reset password'  # this is the title for password resets where the account doesn't exist yet. We may need to test this too ultimately
+    if re.search(activate_subject, bodyContent) is not None:
+        return parseBody(bodyContent, activate_subject, 'activation', 'signup\/confirm')
+
+    if re.search(reset_password_subject, bodyContent) is not None:
+        return parseBody(bodyContent, reset_password_subject, 'passwordreset', 'forgot-password\/reset')
+
+    if re.search(reset_password_no_account_subject, bodyContent) is not None:
+        print("Found Password reset for a non-existent account. This shouldn't happen during tests, one explanation can be running password reset test before test that signs user up")
+        return write_unrecognized_file(s3Key,bodyContent,'noaccountpasswordreset')
+
+    # handle other emails. Ultimately, we should be testing these other emails as well
+    print("Found an email that is not an Activate or Password reset. Don't know what to do with it")
+    write_unrecognized_file(s3Key,bodyContent,'unrecognized')
```

Comment:
You've got a lot of spacing consistency issues in this file in the code & function def's

Reply:
fixed a bunch now